mlserver/batch_processing.py

```diff
@@ -371,10 +502,11 @@ async def process_batch(
 
     for consumer in consumers:
         consumer.cancel()
+    await asyncio.gather(*consumers, return_exceptions=True)
 
     finalizer.cancel()
-
-    await asyncio.gather(*consumers, finalizer, return_exceptions=True)
+    processed_instances = await finalizer
```

Comment:
Good catch! I think `asyncio.gather` doesn't guarantee the order.

Reply:
It actually did not matter here as this was after:
- `await` on queues' `join` returned
- these coroutines where cancelled 

I only made this change to actually get the output of of `finalizer` so I can print number of processed instances